packages/analytics-browser/src/browser-client.ts

```diff
@@ -83,10 +84,21 @@ export class AmplitudeBrowser extends AmplitudeCore implements BrowserClient {
 
     // Step 3: Set session ID
     // Priority 1: `options.sessionId`
-    // Priority 2: last known sessionId from user identity storage
+    // Priority 2: sessionId from url if it's Number
+    // Priority 3: last known sessionId from user identity storage
     // Default: `Date.now()`
     // Session ID is handled differently than device ID and user ID due to session events
-    this.setSessionId(options.sessionId ?? this.config.sessionId ?? Date.now());
+    const queryParams = getQueryParams();
+    this.setSessionId(
```

Comment:
Can we add some helper function for this?
```suggestion
const querySessionId =   (queryParams.ampSessionId && isValidNumber(queryParams.ampSessionId) ? Number(queryParams.ampSessionId) : undefined;
          
 this.setSessionId(
      options.sessionId ??
      querySessionId ??
      this.config.sessionId ??
      Date.now(),
    );```

Reply:
I had a sync with Justin on this 

option 1:
```
_init(){    
    const queryParams = getQueryParams();
    this.setSessionId(
        options.sessionId ??
        (queryParams.ampSessionId ?
            (Number.isNaN(Number(queryParams.ampSessionId)) ?
                    undefined : Number(queryParams.ampSessionId)
            ) : undefined
        ) ??
        this.config.sessionId ??
        Date.now(),
    );
}
```

option 2:
```
_init(){
    this.setSessionId(this._getSessionId());
}

private _getSessionId(options: BrowserOptions) {
    const queryParams = getQueryParams();
    if (options.sessionId) return options.sessionId;
    if (queryParams.ampSessionId && !Number.isNaN(queryParams.ampSessionId)) {
      return Number.isNaN(queryParams.ampSessionId)
    }
    return this.config.sessionId || Date.now(); 
  }
```

I go with option 1 to avoid introducing one more helper method and existing comments can also help on explaining what is going on. I also use indents to make the hierarchy more clear.

Any other pros of option 2 adding a helper function?

Reply:
I would prefer to at least extract all the following into a variable to make it easier to read.
```
        (queryParams.ampSessionId ?
            (Number.isNaN(Number(queryParams.ampSessionId)) ?
                    undefined : Number(queryParams.ampSessionId)
            ) : undefined
        ) ?
```

Reply:
I see what you mean. Thanks Alyssa for pointing it out. That's a better solution! I updated at https://github.com/amplitude/Amplitude-TypeScript/pull/753/commits/5eb9495326287700ee6aca664ea87525b3b0591d.